### DISCUSSION TITLE
Access mastodon's ruby server locally

### DISCUSSION THREAD

**Discussion by felixriehm** [mastodon/mastodon#18944] [Q&A] [answered]
I'm currently trying to setup mastodon with docker. I use the docker-compose.yml of the official git repo. If i try to access the ruby server with any ip address it gets blocked

    mastodon-web-1        | [ActionDispatch::HostAuthorization::DefaultResponseApp] Blocked host: 0.0.0.0
    mastodon-web-1        | [ActionDispatch::HostAuthorization::DefaultResponseApp] Blocked host: 127.0.0.1
    mastodon-web-1        | [ActionDispatch::HostAuthorization::DefaultResponseApp] Blocked host: localhost

All containers run fine. Including (which i try to access to see the web interface):

    mastodon-web-1        | [9] Puma starting in cluster mode...
    mastodon-web-1        | [9] * Puma version: 5.6.4 (ruby 3.0.3-p157) ("Birdie's Version")
    mastodon-web-1        | [9] *  Min threads: 5
    mastodon-web-1        | [9] *  Max threads: 5
    mastodon-web-1        | [9] *  Environment: production
    mastodon-web-1        | [9] *   Master PID: 9
    mastodon-web-1        | [9] *      Workers: 2
    mastodon-web-1        | [9] *     Restarts: (✔) hot (✖) phased
    mastodon-web-1        | [9] * Preloading application
    mastodon-web-1        | [9] * Listening on http://0.0.0.0:3000
    mastodon-web-1        | [9] Use Ctrl-C to stop
    mastodon-web-1        | [9] - Worker 0 (PID: 12) booted in 0.0s, phase: 0
    mastodon-web-1        | [9] - Worker 1 (PID: 17) booted in 0.0s, phase: 0

I have tried to set the `RAILS_ENV` variable to 'development' inside the `dockerfile` since it might have something to do with the [ruby host configuration](https://guides.rubyonrails.org/configuring.html#configuring-middleware). But then i get  

    => ERROR [stage-1  9/10] RUN cd ~ &&  OTP_SECRET=precompile_placeholder SECRET_KEY_BASE=precompile_placeholder rails assets:precompile &&  yarn cache clean       2.2s 
    #0 2.111 rails aborted!                                                                                                                                                 
    #0 2.113 NameError: uninitialized constant Annotat
    ...

when i rerun `docker compose build`. 

Does someone know how i can fix this?

**Comment by ClearlyClaire** [comment, marked as answer]
Mastodon indeed instructs Rails to reject requests with an unexpected `Host`. This means you need to access your Mastodon instance using its configured hostname (typically, `LOCAL_DOMAIN`). If you have set `WEB_DOMAIN`, this value is also accepted (do **not** set it if you don't understand what it means). Alternatively, you can add domains to `ALTERNATE_DOMAINS` for Mastodon to accept requests targeting other domains.

The docker configuration is not intended for `RAILS_ENV=development`, and the error you are encountering suggests the development dependencies haven't been installed (`bundle install --with development`).
